### DISCUSSION TITLE
Extra fields design

### DISCUSSION THREAD

**Discussion by mortii** [mortii/anki-morphs#99] [Ideas]
As discussed in #95

**Comment by mortii** [comment]
I'm not totally sure what I want this to look like. I'm thinking just checkboxes? something like:

- [ ] Add 'am-unknowns' field to your note type?
- [ ] Add 'am-difficulty' field to your note type?
- etc

Edit:

![Screenshot from 2023-12-13 18-45-36](https://github.com/mortii/anki-morphs/assets/15674619/51828954-140f-4d7e-8ae1-881fbb896fb7)

A table view is the probably not the ideal view for this kind of thing...

Edit 2:

![Screenshot from 2023-12-13 21-16-54](https://github.com/mortii/anki-morphs/assets/15674619/848de0e1-3472-4609-bacf-9c957c9b1410)


thoughts @aleksejrs ?

**Reply by aleksejrs** [reply]
>> The field will be filled automatically if you added it.
>
> Whaaaat? I always made them manually. 

I said "filled", not "added", LOL.

**Reply by mortii** [reply]
@aleksejrs I misunderstood you in the best possible way, lol. This is so much better than how morphman does it, and better than the previous way ankiimorphs did it.

**Comment by mortii** [comment]
I have settled on this:

![Screenshot from 2023-12-14 12-57-40](https://github.com/mortii/anki-morphs/assets/15674619/0e9b3e16-c911-44f8-bf9f-376d26a42f51)

**Reply by aleksejrs** [reply]
It could check (and remember the result for each note type) for the contents of the fields of a note type to see if the field could have been created for something else, because "am-highlighted" seems like it could be created by a user.

**Reply by mortii** [reply]
@aleksejrs true, all of these could have already been created, but I think the likelihood is fairly low, even for 'am-highlighted'. If someone creates a bug report where they experienced that problem then it is definitely something we can consider implementing :+1:

**Comment by mortii** [comment]
One potential problem with the table view:

![image](https://github.com/mortii/anki-morphs/assets/15674619/42c27c41-24ed-4ae9-a22b-070c62a09993)

is that it does not scale well, if we decide to add more extra fields then it will get too wide very quickly.

**Comment by mortii** [comment]
I don't like the look of this, but it's probably the way to go:

![image](https://github.com/mortii/anki-morphs/assets/15674619/0ca0151b-8499-43c8-ac3f-47e51f87c18c)
